src/memefactory/server/syncer.cljs

```diff
@@ -301,15 +301,12 @@
         keys->values (->> #{"challengePeriodDuration" "commitPeriodDuration" "revealPeriodDuration" "deposit"
                             "challengeDispensation" "voteQuorum" "maxTotalSupply" "maxAuctionDuration"}
                           (map (fn [k] (when-let [v (records->values (web3/sha3 k))] [k v])))
-                          (into {}))
-        rows (reduce (fn [res [k v]]
-                       (conj res {:initial-param/key k
-                                  :initial-param/db address
-                                  :initial-param/value (bn/number v)
-                                  :initial-param/set-on timestamp}))
-                     []
-                     keys->values)]
-    (js/Promise.resolve (db/insert-initial-params! rows))))
+                          (into {}))]
+    (js/Promise.resolve (doseq [[k v] keys->values]
```

Comment:
no need for a loop here - upsert should work with multiple rows too - lets limit the number of db operations

Reply:
I'm not worried about perf here since this is only going to be called on parameter change apply so once in a while, and we are only talking about the order of 10 rows always. But it isn't hard to change neither

Reply:
It's just that I was very careful to avoid unnecessary db operations when re-designing how we do persistance, introduced upserts instead of if-exists patterns and bulk inserts wherever possible. This is more performant and will pay off if we decide to go for a regular database. Not to mention that here you get it for free, you have the code that turns the data into insertable rows, you can just use it.